frontend/src/features/admin-form/settings/components/EndPageSettingsSection/EndPageSettingsInput.tsx

```diff
@@ -0,0 +1,111 @@
+import { useCallback, useMemo } from 'react'
+import { useForm } from 'react-hook-form'
+import { FormControl, Stack } from '@chakra-ui/react'
+import validator from 'validator'
+
+import { FormEndPage } from '~shared/types'
+
+import FormErrorMessage from '~components/FormControl/FormErrorMessage'
+import FormLabel from '~components/FormControl/FormLabel'
+import Input from '~components/Input'
+import Textarea from '~components/Textarea'
+
+import { useMutateFormPage } from '~features/admin-form/common/mutations'
+
+export const EndPageSettingsInput = ({
+  title,
+  paragraph,
+  buttonLink,
+  buttonText,
+}: FormEndPage): JSX.Element => {
+  const { mutateFormEndPage } = useMutateFormPage()
+
+  const defaultParagraph = useMemo(() => paragraph ?? '', [paragraph])
+  const defaultButtonLink = useMemo(() => buttonLink ?? '', [buttonLink])
+
+  const {
+    register,
+    formState: { errors },
+    resetField,
+    handleSubmit,
+  } = useForm<FormEndPage>({
+    mode: 'onChange',
+    defaultValues: {
+      title: title,
+      paragraph: defaultParagraph,
+      buttonText: buttonText,
+      buttonLink: defaultButtonLink,
+    },
+  })
+
+  const handleUpdateEndPage = useCallback(() => {
+    return handleSubmit((endPage) => {
+      if (
+        endPage.title === title &&
+        endPage.buttonText === buttonText &&
+        endPage.buttonLink === defaultButtonLink &&
+        endPage.paragraph === defaultParagraph
+      ) {
```

Comment:
these conditional checks looks a little off. the benefit of `react-hook-form` is that once `handleSubmit` is called, we know that validation has passed. Only valid inputs should be passed in (and thus has no need for these checks).

but i see that there are no validation on the fields yet, will flag in later comments.

Reply:
This check here prevents the case of no changes in any `endPage` data when a user clicks between the different inputs still causing a `PUT` request to still be sent. It's also because of the `onBlur` updates. Will clarify if it can be changed to a submit button